core/src/main/java/org/apache/calcite/sql/type/SqlTypeFactoryImpl.java

```diff
@@ -126,6 +126,17 @@ public SqlTypeFactoryImpl(RelDataTypeSystem typeSystem) {
     return canonize(newType);
   }
 
+  /**
+   * Creates a TZAware SQL type. This is a Timestamp with particular Timezone information.
+   *
+   * @param tzInfo    The timezone information
+   * @return canonical TZAware SQL type descriptor
+   */
+  @Override public RelDataType createTZAwareSqlType(BodoTZInfo tzInfo) {
+    RelDataType newType = new TZAwareSqlType(typeSystem, tzInfo, false);
+    return canonize(newType);
```

Comment:
I assume canonize just registers the type with the type factory if it hasn't been already? or something similar?